biz.aQute.bndlib/src/aQute/bnd/osgi/Analyzer.java

```diff
@@ -2562,6 +2562,9 @@ public String _packages(String... args) throws Exception {
 							throw new IllegalArgumentException("Not enough arguments in ${packages} macro");
 						accept = pkgInfo != null && pkgInfo.is(Clazz.QUERY.ANNOTATED, instr, this);
 						break;
+					case EXPORTED :
+						accept = entry.getValue().getVersion() != null;
```

Comment:
Is this really the way to know the package is exported? Couldn't a non-exported package have a version? Perhaps we need a different macro which is invoked on the exported package set. This macro is invoked on the contained package set.